### PR TITLE
[Spark load][Fe 4/6] Add hive external table and update hive table syntax in loadstmt

### PR DESCRIPTION
**Pull request by wyb**
#3787 
#3433

### CODE REVIEW

File: fe/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -1244,6 +1244,15 @@ data_desc ::=
         RESULT = new DataDescription(tableName, partitionNames, files, colList, colSep, fileFormat,
         columnsFromPath, isNeg, colMappingList, whereExpr);
     :}
+    | KW_DATA KW_FROM KW_TABLE ident:srcTableName
+    opt_negative:isNeg
+    KW_INTO KW_TABLE ident:tableName
+    opt_partition_names:partitionNames
+    opt_col_mapping_list:colMappingList
```
[the remark this answers is not in the document]
Comment (imay) [reply fe/src/main/cup/sql_parser.cup:1251]: @morningman 
What you suggested seems more flexible. however I think it is a more complex feature.
In my opinion, we can only support table directlly to fulfill 80% user requirements. 
And we can implement what your suggest later when it is needed.

And we can keep the same with load file in column mapping by column_lists and SET clause.

File: fe/src/main/java/org/apache/doris/load/BrokerFileGroup.java [hunk 2]
```diff
@@ -165,6 +170,33 @@ public void parse(Database db, DataDescription dataDescription) throws DdlExcept
...
+            String srcTableName = dataDescription.getSrcTableName();
+            // src table should be hive table
+            Table srcTable = db.getTable(srcTableName);
+            if (srcTable == null) {
+                throw new DdlException("Unknown table " + srcTableName + " in database " + db.getFullName());
+            }
+            if (!(srcTable instanceof HiveTable)) {
+                throw new DdlException("Source table " + srcTableName + " is not HiveTable");
+            }
+            // src table columns should include all columns of loaded table
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/src/main/java/org/apache/doris/load/BrokerFileGroup.java:184]: Support later

File: fe/src/main/cup/sql_parser.cup (hunk 1 above)
[the remark this answers is not in the document]
Comment (wyb) [reply fe/src/main/cup/sql_parser.cup:1251]: Support later

File: fe/src/main/java/org/apache/doris/analysis/FromClause.java [hunk 3]
```diff
@@ -72,6 +78,39 @@ public int compare(TableRef tableref1, TableRef tableref2) {
...
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+
+            Database db = analyzer.getCatalog().getDb(dbName);
+            if (db == null) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
+            }
+
+            String tblName = tableName.getTbl();
+            Table table = db.getTable(tblName);
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/analysis/FromClause.java:104]: Lock db before getting table

Comment (wyb) [reply fe/src/main/java/org/apache/doris/analysis/FromClause.java:104]: Lock is acquired in StmtExecutor.analyze function.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**